pkg/scaling/scale_handler.go

```diff
@@ -611,3 +607,171 @@ func (h *scaleHandler) getScaledObjectState(ctx context.Context, scaledObject *k
 
 	return isScaledObjectActive, isScalerError, metricsRecord, nil
 }
+
+type scalerMetrics struct {
+	queueLength float64
+	maxValue    float64
+	isActive    bool
+}
+
+// / --------------------------------------------------------------------------- ///
+// / ----------             ScaledJob related methods               --------- ///
+// / --------------------------------------------------------------------------- ///
+
+// getScaledJobMetrics returns metrics for specified metric name for a ScaledJob identified by its name and namespace.
+// It could either query the metric value directly from the scaler or from a cache, that's being stored for the scaler.
+func (h *scaleHandler) getScaledJobMetrics(ctx context.Context, scaledJob *kedav1alpha1.ScaledJob) []scalerMetrics {
+	// TODO this loop should be probably done similar way the ScaledObject loop is done
+	cache, err := h.GetScalersCache(ctx, scaledJob)
+	if err != nil {
+		log.Error(err, "error getting scalers cache", "scaledJob.Namespace", scaledJob.Namespace, "scaledJob.Name", scaledJob.Name)
+		return nil
+	}
+	var scalersMetrics []scalerMetrics
+	scalers, _ := cache.GetScalers()
+	for i, s := range scalers {
+		var queueLength float64
+		var targetAverageValue float64
+		isActive := false
+		maxValue := float64(0)
+		scalerType := fmt.Sprintf("%T:", s)
+
+		scalerLogger := log.WithValues("ScaledJob", scaledJob.Name, "Scaler", scalerType)
+
+		metricSpecs := s.GetMetricSpecForScaling(ctx)
+
+		// skip scaler that doesn't return any metric specs (usually External scaler with incorrect metadata)
+		// or skip cpu/memory resource scaler
+		if len(metricSpecs) < 1 || metricSpecs[0].External == nil {
+			continue
+		}
+
+		// TODO here we should probably loop through all metrics in a Scaler
```

Comment:
could you please address this TODO?

Reply:
ditto.

I think it's looping through all metrics now.
Could you share more detail?

Reply:
Ah, I can see it is fixed, then please remove the TODO

Reply:
Done.